1-mandatory/3-slideshow/slideshow.js

```diff
@@ -1 +1,126 @@
+        const h2Row = document.createElement('h2');
+        h2Row.style.setProperty('--i', cursorSpeed[i])
+        h2Row.style.setProperty('--shadow-color', shColor)
+        Array.from(Array(10).keys()).forEach((j) => {
+            const span = document.createElement('span');
+            span.innerText = 'CodeYourFuture'
+            if (j % 2 !== 0) span.style.webkitTextStroke = `2px ${contourColor}`;
+            h2Row.appendChild(span);
+        })
+        document.getElementById('animatedText').appendChild(h2Row);
+        document.body.style.backgroundColor = bgColor;
+    })
+}
+const images = [
+    {url: 'https://images.unsplash.com/photo-1513635269975-59663e0ac1ad?ixlib=rb-1.2.1&ixid=MnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8&auto=format&fit=crop&w=1170&q=80', title: 'The View from The Shard, London, United Kingdom', author: 'Benjamin Davies'},
+    {url: 'https://images.unsplash.com/photo-1533929736458-ca588d08c8be?ixlib=rb-1.2.1&ixid=MnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8&auto=format&fit=crop&w=1170&q=80', title: 'Tower Bridge, London, United Kingdom', author: 'Charles Postiaux'},
+    {url: 'https://images.unsplash.com/photo-1486299267070-83823f5448dd?ixlib=rb-1.2.1&ixid=MnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8&auto=format&fit=crop&w=1171&q=80', title: 'Big Ben, London, United Kingdom', author: 'Marcin Nowak'},
+    {url: 'https://images.unsplash.com/photo-1520986606214-8b456906c813?ixlib=rb-1.2.1&ixid=MnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8&auto=format&fit=crop&w=1170&q=80', title: 'Big Ben', author: 'Lucas Davies'},
+    {url: 'https://images.unsplash.com/photo-1513026705753-bc3fffca8bf4?ixlib=rb-1.2.1&ixid=MnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8&auto=format&fit=crop&w=1170&q=80', title: 'London East Side from The Shard', author: 'Giammarco'},
+]
+let i = 0;
+let timeInterval = 2000;
+const imgContainer = document.getElementById('img-container');
+const next = document.getElementById('next');
+const prev = document.getElementById('previous');
+const aNext = document.getElementById('auto-forward');
+const aPrev = document.getElementById('auto-back');
+const stop = document.getElementById('stop');
+let slideshowState;
+let darkMode = false;
+
+const DIRECTIONS = {
+    NEXT: 1, 
+    PREV: -1
+}
+let currDirection = DIRECTIONS.NEXT;
+
+const displayImage = () => {
+    const imgEl = document.getElementById('img');
+    imgEl.setAttribute('src', images[i].url);
+    document.getElementById('description').innerText = `${images[i].title} by ${images[i].author}`
+}
+
+const changeImage = () => {
+    if (currDirection === DIRECTIONS.NEXT) {
+        i = i === images.length - 1 ? 0 : i + 1;
+    } else {
+        i = i === 0 ? images.length - 1 : i - 1;
+    }
+    displayImage()
+}
+
+next.addEventListener('click', () => {
+    currDirection = DIRECTIONS.NEXT;
+    changeImage()
+})
+
+prev.addEventListener('click', () => {
+    currDirection = DIRECTIONS.PREV;
+    changeImage()
+});
```

Comment:
When you change the image to the previous or next, you should also reset the timeout, otherwise, you won't have a consistent pause between pictures.